components-core/src/main/java/com/adyen/checkout/components/PaymentComponentState.java

```diff
@@ -8,15 +8,30 @@
 
 package com.adyen.checkout.components;
 
+import android.os.Parcel;
+import android.os.Parcelable;
+
 import androidx.annotation.NonNull;
 
 import com.adyen.checkout.components.model.payments.request.PaymentComponentData;
 import com.adyen.checkout.components.model.payments.request.PaymentMethodDetails;
+import com.adyen.checkout.core.util.ParcelUtils;
 
 /**
  * The current state of a PaymentComponent.
  */
-public class PaymentComponentState<PaymentMethodDetailsT extends PaymentMethodDetails> {
+public class PaymentComponentState<PaymentMethodDetailsT extends PaymentMethodDetails> implements Parcelable {
+
+    @NonNull
+    public static final Parcelable.Creator<PaymentComponentState> CREATOR = new Parcelable.Creator<PaymentComponentState>() {
```

Comment:
Android Studio is complaining about the raw type parameter, so maybe make it `PaymentComponentData<? extends PaymentMethodDetails>`

Reply:
Hmm right. But I will still get this warning in `return new PaymentComponentState(in);` and `return new PaymentComponentState[size];`